### DISCUSSION TITLE
Issues with SHAP DeepExplainer and SELU Activation in TensorFlow

### DISCUSSION THREAD

**Discussion by MZA85** [shap/shap#3499] [General] [answered]
Hello all,

I'm currently working with SHAP version 0.44 to interpret a deep learning model built using TensorFlow 2.13, running on Python 3.8. My model utilizes the SELU activation function extensively across various layers. While attempting to generate explanations for my model's predictions using SHAP's DeepExplainer, I encountered difficulties that seem to stem specifically from the use of SELU activation.
I don't have any issues with other activation functions, and other explainers such as GradientExplainer and KernelExplainer.

This is the error which I got:  
`LookupError: gradient registry has no entry for: shap_Selu`

![image](https://github.com/shap/shap/assets/159890939/32af9816-2270-4435-a435-0c51f577139b)

Does anyone has similar experience & issue?
Could you please help me to solve this issue?

Thank you very much.

**Comment by CloseChoice** [comment, marked as answer]
Hi,
the DeepExplainer needs a predefined entry for each supported layer, e.g. see https://github.com/shap/shap/blob/master/shap/explainers/_deep/deep_tf.py#L691-L756. 
Could you try to add a line
`op_handlers["Selu"] = nonlinearity_1d(0)` there? Would be great to report your findings back here so that we can add it to the package if it works

**Comment by MZA85** [comment]
Hi,

Thanks so much. It works perfectly.

**Reply by CloseChoice** [reply]
could you please mark my answer as correct, so that this discussion can be closed? I created an issue for this: https://github.com/shap/shap/issues/3502. Feel free to pick it up if you would like to contribute this feature
